lang/java/avro/src/main/java/org/apache/avro/io/JsonDecoder.java

```diff
@@ -453,11 +445,12 @@ public Symbol doAction(Symbol input, Symbol top) throws IOException {
       Symbol.FieldAdjustAction fa = (Symbol.FieldAdjustAction) top;
       String name = fa.fname;
       if (currentReorderBuffer != null) {
-        List<JsonElement> node = currentReorderBuffer.savedFields.get(name);
-        if (node != null) {
+        TokenBuffer tokenBuffer = currentReorderBuffer.savedFields.get(name);
```

Comment:
Can we wrap a try-with-resources here as well?

Reply:
I think the try-with-resource is effectively a no-op here, since the `.close()` method seems to make the buffer [read-only](http://fasterxml.github.io/jackson-core/javadoc/2.3.0/com/fasterxml/jackson/core/JsonGenerator.html?is-external=true#close()). After insertion, the buffer should be read-only for the rest of its lifetime.